### ISSUE TITLE
DNS Serivice Discovery: vminsert/vmselect when adding new storage

### ISSUE THREAD

**Issue by tenmozes** [VictoriaMetrics/VictoriaMetrics#3329]
**Describe the bug**
vminsert/vmselect  panics when adding new storage node using DNS SRV discovery

**Stack Trace**
```
VictoriaMetrics/lib/logger/flag.go:20     -storageNode="dns+srv:vmstorage-secret.consul"
2022-11-08T18:41:34.081963284Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/main.go:84 initializing netstorage for storageNodes [dns+srv:vmstorage*secret*.consul]...
2022-11-08T18:41:34.082452484Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/netstorage/storage_nodes_watcher.go:22 discovered vmstorage nodes: ["0a66eafb.addr.*secret*.consul.:21060" "0a66ec24.addr.*secret*.consul.:26579" "0a66efb9.addr.*secret*.consul.:26795"]
2022-11-08T18:41:34.082843267Z instance-vminsert 17 info VictoriaMetrics/lib/memory/memory.go:42 limiting caches to 966367641 bytes, leaving 107374183 bytes to the OS according to -memory.allowedPercent=90
2022-11-08T18:41:34.082866132Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/main.go:101    successfully initialized netstorage in 0.001 seconds
2022-11-08T18:41:34.082957634Z instance-vminsert 17 info VictoriaMetrics/lib/httpserver/httpserver.go:96 starting http server at http://ip:port/
2022-11-08T18:41:34.082971577Z instance-vminsert 17 info VictoriaMetrics/lib/httpserver/httpserver.go:97 pprof handlers are exposed at http://ip:port/debug/pprof/
2022-11-08T18:41:34.289615258Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/netstorage/netstorage.go:265   successfully dialed -storageNode="0a66eafb.addr.*secret*.consul.:21060"
2022-11-08T18:41:34.290027883Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/netstorage/netstorage.go:265   successfully dialed -storageNode="0a66ec24.addr.*secret*.consul.:26579"
2022-11-08T18:41:34.290871514Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/netstorage/netstorage.go:265   successfully dialed -storageNode="0a66efb9.addr.*secret*.consul.:26795"
2022-11-08T19:28:02.084154592Z instance-vminsert 17 info VictoriaMetrics/app/vminsert/netstorage/storage_nodes_watcher.go:62 Updating the list of vmstorage nodes: added=["0a66e9e1.addr.*secret*.consul.:29539" "0a66ee09.addr.*secret*.consul.:26301"], deleted=[], discovered addrs=["0a66e9e1.addr.*secret*.consul.:29539" "0a66eafb.addr.*secret*.consul.:21060" "0a66ec24.addr.*secret*.consul.:26579" "0a66ee09.addr.*secret*.consul.:26301" "0a66efb9.addr.*secret*.consul.:26795"]
2022-11-08T19:28:02.189157539Z instance-vminsert 17 panic: runtime error: index out of range [3] with length 3
2022-11-08T19:28:02.189175384Z instance-vminsert 17
2022-11-08T19:28:02.18918754Z instance-vminsert 17 goroutine 78354 [running]:
2022-11-08T19:28:02.18919341Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper.func1()
2022-11-08T19:28:02.189199407Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:229 +0x65
2022-11-08T19:28:02.189207053Z instance-vminsert 17 panic({0x8d8000, 0xc0015c8090})
2022-11-08T19:28:02.189211936Z instance-vminsert 17  runtime/panic.go:890 +0x262
2022-11-08T19:28:02.189217904Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/netstorage.(*InsertCtx).WriteDataPointExt(0xc00025c400?, 0x0?, {0xc00025c400?, 0x0?, 0xc00058e000?}, 0xe?, 0x10?)
2022-11-08T19:28:02.189226259Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/netstorage/insert_ctx.go:129 +0x1fd
2022-11-08T19:28:02.189241354Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.insertRows(0xc00149b3c8?, {0xc001dee000, 0xa, 0x0?}, {0x0, 0x0, 0xc001346140?})
2022-11-08T19:28:02.189248333Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite/request_handler.go:75 +0x7f6
2022-11-08T19:28:02.189254378Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler.func1.1({0xc001dee000?, 0xc001cbc000?, 0x15a5?})
2022-11-08T19:28:02.18926011Z instance-vminsert 17   github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite/request_handler.go:33 +0x3a
2022-11-08T19:28:02.189265273Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/promremotewrite.ParseStream({0x7f70277bea98?, 0xc0001c8580?}, 0xc00149b5c8)
2022-11-08T19:28:02.189270601Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/lib/protoparser/promremotewrite/streamparser.go:57 +0x5bf
2022-11-08T19:28:02.189275713Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler.func1()
2022-11-08T19:28:02.189280921Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite/request_handler.go:32 +0x86
2022-11-08T19:28:02.189287529Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter.Do(0xc00149b6b8)
2022-11-08T19:28:02.189291762Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/lib/writeconcurrencylimiter/concurrencylimiter.go:37 +0x222
2022-11-08T19:28:02.189301692Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite.InsertHandler(0xc0003826a0, 0xc001044400)
2022-11-08T19:28:02.189311038Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/promremotewrite/request_handler.go:31 +0x8a
2022-11-08T19:28:02.189317602Z instance-vminsert 17 main.requestHandler({0xa180d8, 0xc00017de30}, 0xc001044400)
2022-11-08T19:28:02.189324555Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/app/vminsert/main.go:207 +0xb16
2022-11-08T19:28:02.189337602Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.handlerWrapper(0xc00009f150, {0xa180d8?, 0xc00017de30}, 0xc001044400, 0x9a19b0)
2022-11-08T19:28:02.189346589Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:337 +0xdcd
2022-11-08T19:28:02.18935514Z instance-vminsert 17 github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver.gzipHandler.func1({0xa18348?, 0xc00104c380?}, 0xc00017de00?)
2022-11-08T19:28:02.189362818Z instance-vminsert 17  github.com/VictoriaMetrics/VictoriaMetrics/lib/httpserver/httpserver.go:197 +0x76
2022-11-08T19:28:02.189371583Z instance-vminsert 17 net/http.HandlerFunc.ServeHTTP(0x0?, {0xa18348?, 0xc00104c380?}, 0x64b4b4?)
2022-11-08T19:28:02.189379075Z instance-vminsert 17  net/http/server.go:2109 +0x2f
2022-11-08T19:28:02.189386237Z instance-vminsert 17 net/http.serverHandler.ServeHTTP({0xa179f8?}, {0xa18348, 0xc00104c380}, 0xc001044400)
2022-11-08T19:28:02.189393631Z instance-vminsert 17  net/http/server.go:2947 +0x30c
2022-11-08T19:28:02.18940473Z instance-vminsert 17 net/http.(*conn).serve(0xc000386640, {0xa188a0, 0xc000a7af00})
2022-11-08T19:28:02.189411033Z instance-vminsert 17  net/http/server.go:1991 +0x607
2022-11-08T19:28:02.189419235Z instance-vminsert 17 created by net/http.(*Server).Serve
2022-11-08T19:28:02.18944205Z instance-vminsert 17   net/http/server.go:3102 +0x4db
2022-11-08T19:28:02.196565027Z instance-vminsert exited with error: exit status 1
```

**Comment by valyala** [comment]
The bug should be fixed in the enterprise branch. The bugfix can be tested by using `vminsert` docker image with the tag `heads-enterprise-cluster-0-g8db45f06d` . The bugfix will be included in the next release of VictoriaMetrics.

**Comment by valyala** [comment]
The panic during [vmstorage discovery](https://docs.victoriametrics.com/Cluster-VictoriaMetrics.html#automatic-vmstorage-discovery) at `vminsert` enterprise has been fixed in the [VictoriaMetrics enterprise v1.83.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the bug as fixed.
